src/devices/Bmm150/Bmm150I2c.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.I2c;
+
+namespace Iot.Device.Bmp180
+{
+    /// <summary>
+    /// Default I2C interface for the Bmm150
+    /// </summary>
+    public class Bmm150I2c : Bmm150I2cBase

```

Comment:
Why is this additional indirection class required? The I2cDevice itself is already intended to be swapped with different implementations

Reply:
Let me answer this one: BMI160 from Bosh include this sensor. To access it, similar to the AKS8963 you need a an abstraction like this one. See https://www.bosch-sensortec.com/products/motion-sensors/imus/bmi160/

Reply:
I know the use case, but why can't the BMI160 directly implement the `I2cDevice` interface (or rather, provide it?). It's actually a kind of I2C hub.

Reply:
it can but just for the raw values. For the calibration, you need to access the BMM150 directly thru the BMI160 unfortunately.

Reply:
I'm sorry,  but I don't get your point. This interface is just a subset of `I2cDevice`. Nothing device-specific in here.

Reply:
To access BMM150 thru BMI160, you need to use BMI160 I2cDevice and its I2C address. So if you dispose it before BMI160, then you'll have issues when disposing BMI160. So it's the reason, the same as for AK8963 that the I2C Device you give should not be disposed before the main device is disposed. I hope it does make more sense.

Reply:
That applies to the `shouldDispose` above, but not to the question here: Why are the abstract classes `Bmm150I2cBase` and it's implementation `Bmm150I2c` required? The former is just a subset of `I2cDevice` and the later is just a forwarding adapter.